### ISSUE TITLE
Add file filtering to FileOpen dialog

### ISSUE THREAD

**Issue by juja256** [fyne-io/fyne#4195]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I am always frustrated with a lack of ability of file filtering in FileOpen dialog, it would be very useful to add this feature to fyne, especially when user have a lot of different files in directory and searching the appropriate one is a bit hard.

### Describe the solution you'd like to see.

It would be great to add entry search widget to FileOpen dialog, somewhat like that:
![Untitled](https://github.com/fyne-io/fyne/assets/2045262/18d14101-1189-4fa8-ad25-3a6907ca5562)

**Comment by juja256** [comment]
For example, when user have to find some file he needs to search all over the files(sorted lexicographically) manually in a directory, it is a bit inconvenient especially when there is a lot of files. System dialogs usually give ability to search file via Ctrl+F or by clicking at search entry: user types some text into entry, after that new file filter will be applied and the list of files in the directory is refreshed.

**Comment by andydotxyz** [comment]
Aha thanks, so this is limiting the results in the current directory not searching across the computer?

**Comment by matwachich** [comment]
So we should call this "file filtering"

**Comment by Jacalz** [comment]
It is worth noting that the search box in Nautilus (Gnome Files) seems to search in the local directory and recursively down into any sub directories when typing something in to it. I think it is a good way to handle it but in that case I wouldn't call it file filtering.

**Comment by juja256** [comment]
Yeah, it would be great to search in the local directory and recursively down into sub-directories.
